lib/templatesimplifier.cpp

```diff
@@ -320,6 +329,38 @@ unsigned int TemplateSimplifier::templateParameters(const Token *tok)
     return 0;
 }
 
+// Same as Token::eraseTokens() but tries to fix up lists with pointers to the deleted tokens.
+void TemplateSimplifier::eraseTokens(Token *begin, const Token *end)
+{
+    if (!begin || begin == end)
+        return;
+
+    while (begin->next() && begin->next() != end) {
+        // check if token has a pointer to it
+        if (begin->next()->hasPointer()) {
+            // check if the token is in a list
+            const std::list<TokenAndName>::iterator it2 = std::find_if(mTemplateInstantiations.begin(),
```

Comment:
Please rename `it2` to `it`

Reply:
Fixed.  This was a leftover from a following patch that also fixes up template declaration pointers. That problem is not encountered in the 2 tests I added.